src/components/text-editor/prosemirror-adapter/mentions/node-schema-extender.ts

```diff
@@ -0,0 +1,35 @@
+import { NodeSpec } from 'prosemirror-model';
+
+export const mention: NodeSpec = {
```

Comment:
Many aspects of this will be agnostic and generic. However for NodeSpec's they will have to be unique and specific to what they hold. 

If we're traversing the Editor State the current Node or group of Nodes needs to tell us exactly what it or they are. So for tags we would create a separate NodeSpec to specify tags.

Reply:
Not sure if I entirely understand the technical aspects that you mentioned. But what I was trying to say is, should this _@-ing_ or _#-ing_ be called "mentioning" and "tagging"? Or can we find more generic names and functionality-agnostic designs for them, such as `at-command`, `hashtag-command` etc… 

If we call it "mention" and create an entire set of user interactions and components for handling mentions (picking a person from a searchable list while typing a piece of text), our work will be easy and straightforward. The consumer's expectations will also be straightforward. They will only use the `@` trigger to implement mentions, where they use the text editor component. 

However, in another context for other consumers, the _@-ing_ may be used to link things together maybe… Or to add _locations_, or _dates_ or whatever the 3rd-party consumers' use-case may be. 

Maybe I'm over complicating… but this was just a question that poped in my head.